Add SingleBarChart to State Details & Add Homepage Electrification Lines

vkoves opened the pull request:
## Overview

Brief description of what this PR does, and why it is needed.

Closes #XXX

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

vkoves commented:
@derekeder - I'd love your early thoughts on this PR. I've added the single bar chart with highlighting as the mocks have except keeping the chart vertical like it is on the homepage. Do you think this is workable, or should we move it to be horizontal like in the mocks? We could alternatively use a pie chart too.

![Screenshot from 2022-03-24 22-10-23](https://user-images.githubusercontent.com/3187531/160046886-cd86a3c3-8911-4af5-a297-0c58b295eea8.png)

derekeder commented:
@vkoves one challenge with the horizontal bar will be fitting the text in the space of the bar, especially if they are smaller like the 15% for buildings.

An earlier version of the wireframes had a 2 column layout with the bar chart being 'sticky' as you scrolled past each section. We could use something like scrollspy to update the chart as we scroll: https://getbootstrap.com/docs/4.0/components/scrollspy/#via-javascript

vkoves commented:
@derekeder - that definitely makes sense, I could give that a try! I will also say that the current graph setup already has issues for some states, like [Wyoming](https://deploy-preview-40--state-decarbonize.netlify.app/wyoming):

![Screenshot from 2022-03-28 12-31-47](https://user-images.githubusercontent.com/3187531/160454554-cbb2a464-d2e3-47ca-bd70-846b49eedbf7.png)

I might be able to fix this by putting the label on top of each graph section, but I'm open to anything

vkoves commented:
@derekeder - I think this is ready for a final review. I think that having a graph that follows you would be a good improvement but should be a separate PR/issue if we indeed want to do it. I'm a little torn, but I think some folks have pointed out that otherwise you're seeing the same graph and over again.

vkoves commented:
I've made a few last improvements based on Slack feedback:

1. If a state has no emissions from power (Vermont and D.C.) I show some custom text and don't show that on the graph. I didn't create the blank text for any other category (since I couldn't find states with 0% building or transport emissions) but the graph should handle 0% values fine
2. I added a new basic "Other" section - we can flesh this out more but I think this is important especially for states like Texas, where Other is a majority of their emissions
3. Added a closing section

![Screenshot from 2022-04-01 23-05-56](https://user-images.githubusercontent.com/3187531/161365599-3570b19d-148c-453d-af3b-eb5aba15ba5a.png)

derekeder commented on src/components/singlebar.js:
```diff
@@ -1,41 +1,271 @@
+        <YAxis domain={['dataMin', 'dataMax']} hide={ true } />
+        {
+          // Only show other bar if it's non-zero
+          emissionsData[BarsConfig.other.key] &&
```
in future PRs we should make this more DRY as the only thing that's really changing is the key we're referencing

derekeder reviewed (approved):
Looks good!

A few notes:
- the 4 bar areas should follow the same order of the sections on the page. If we stick with the sections, it should be Buildings, Transit, Power, Other
- We should go with the updating sticky bar chart on the left with this layout. We're repeating the same bar too many times

vkoves commented:
@derekeder - good points, I've created #46 to address both of those. On mobile, I'm assuming we still need the separate graphs, since a column layout won't fit
